joomla/joomla-cms
# [5.0] Cassiopeia Fontstacks

Currently cassiopeia offers three font configuration settings all with their limitations

1. none - boring generic system font
2. Roboto Local - slow
3. Google Fonts - privacy

Using a system font stack is always the best. It is fast, there is nothing to download, there is no layout shift (google pagespeed hates that).

The problem with a system font stack is that every system has different fonts available so you need to be careful in defining it. Thats why the current system font stack from Bootstrap that is used in Cassiopeia is so boring and safe and why people choose google fonts etc.

This PR adds 14 modern system font stacks from @danklammer

> System font stack CSS organized by typeface classification for every modern OS
The fastest fonts available. No downloading, no layout shifts, no flashes — just instant renders.

You can see them in action at https://modernfontstacks.com/ and helpfully see which font in each stack your own system will use.

This was developed for my own sites and I am offering it here as I am certain that it will be useful to all. Especially to those people who don't realise they can have great font choices with Joomla without impacting privacy or performance.

The base branch for this PR is 4.3 simply because that's what is developed on. I leave it to maintainers to decide if and when it can be merged.

It is fully backwards compatible
I created language strings for each font stack but I suspect that most translation teams will chose not to translate them.

### Testing Instructions
In the template style options for Cassiopeia there is now an additional option in the Fonts Scheme select
![image](https://user-images.githubusercontent.com/1296369/225268383-445620de-0e61-4a01-b485-e5182fabf750.png)

Once the System Fonts option is selected there are two new lists available to chose both a body font stack and a heading font stack. There is also an additional line added to the Notes

![image](https://user-images.githubusercontent.com/1296369/225269211-9553e5cf-6f2f-4356-96c9-f72c06e5f7fa.png)

Once you have selected your stack and saved the style you can test it on the front end.
Viewing the source you should now see
![image](https://user-images.githubusercontent.com/1296369/225269725-4436f06f-33fb-4339-b569-c2c6f69142ef.png)

Inspecting the css you should see
![image](https://user-images.githubusercontent.com/1296369/225269912-fb41abbf-c75a-4644-bc88-70a67953b1fd.png)

### Link to documentations
If/when accepted I will update the docs - who knows which site it will be on by then

Please select:
